include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
+		const Gaffer::StringPlug *colorSpaceMetadataNamePlug() const;
+
 		virtual void affects( const Gaffer::Plug *input, AffectedPlugsContainer &outputs ) const;
 
 		static size_t supportedExtensions( std::vector<std::string> &extensions );
 
+		typedef boost::function<const std::string ( const IECore::CompoundData * )> DefaultColorSpaceFunction;	
+		static void registerDefaultColorSpace( DefaultColorSpaceFunction f );
```

Comment:
> ideally we would have the exact same function used in both case, but as you can see from my default implementation that doesnt quite work ( dataType name not consistent between read and write ).

I think that might be a bug in OpenImageIOReader's dataType computation though. It seems to report floating point tiffs as uint32 rather than float. I created one using `oiiotool --create 4x4 3 -o float.tif` if you want to try yourself. It'd be good to add some test cases for the dataType plug since it will be so important to this mechanism.